src/test/unit/mcmc/hmc/hamiltonians/diag_e_metric_test.cpp

```diff
@@ -52,6 +55,9 @@ TEST(McmcDiagEMetric, gradients) {
   z.q = q;
   z.p.setOnes();
 
+  Eigen::VectorXd mask(q.size());
+  mask << 1, 0, 1, 1, 1, 1, 1, 1, 1, 0, 0;
```

Comment:
Should have a test that makes sure the masking zeros out the right elements in `dtau_dp`, `dphi_dq`, and `sample_p`.